lang/python/wiredtiger/init.py

```diff
@@ -47,6 +52,32 @@
 me = sys.modules[__name__]
 sys.path.append(os.path.dirname(__file__))
 
+# Find the sanitizer environment variable.
+# FIXME-WT-13237: Rename these testutil flags
+if os.environ.get("TESTUTIL_TSAN") == "1":
+    import subprocess
+
+    # FIXME-WT-13143 We assume TSan is only compatible with clang here. This may change in the future.
+    command = "clang --print-file-name libtsan.so.0"
+    find_tsan_so = subprocess.run(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    tsan_so_path = find_tsan_so.stdout.strip()
+    if not os.path.isfile(tsan_so_path):
+        print("Cannot find tsan lib")
+        exit(1)
+
+    ld_preload = os.environ.get("LD_PRELOAD")
+    if not ld_preload:
+        os.environ["LD_PRELOAD"] = tsan_so_path
+        restart_python()
+    elif tsan_so_path not in os.environ["LD_PRELOAD"]:
+        os.environ["LD_PRELOAD"] += f":{tsan_so_path}"
+        restart_python()
```

Comment:
This runs into the issues mentioned in [this comment](https://github.com/wiredtiger/wiredtiger/pull/10839#discussion_r1669705705). 
If we set `LD_PRELOAD` in evergreen.yml it can interfere with command line tools so it needs to be set on Python load.

Reply:
I am not fond of resetting the python for LD_PRELOAD, but I don't to hold this PR because of it. It feels quite hacky and very specific to TSAN variants. I will create a ticket to investigate whether we can find an alternative method.